Automattic/woocommerce-payments
# Check for valid deadline when setting Update Business Details task

Fixes #4846

#### Changes proposed in this Pull Request



* Check whether we have a valid `currentDeadline` when setting the update business details task. 
* Don't show the task if we don't have a `currentDeadline` (since there's no action for the user to take right now).
* Also fixed a bug where we were not checking the proper key when checking for pastDue requirements, so the other task that would appear should work properly now (previously, it would never have been displayed).







#### Testing instructions





**Prerequisite**

There's a few tests here for showing the task based on various conditions. To test properly, you'll need to do the following before every test:

* Update your local server code to return `account_status`, `current_deadline` and `has_overdue_requirements` to the values specified in the test.
* Refresh the account cache using the dev tools.

**Tests**

Testing with a `restricted_soon` account with a current deadline.

* Update `account_status` to `restricted_soon`, and `current_deadline` to a date in the future.
* Check the `Payments > Overview` page. The task should appear and look like this:

<img width="728" alt="Screenshot 2022-11-29 at 11 22 33" src="https://user-images.githubusercontent.com/11770181/204516437-6fdbf33a-95a7-4f11-be45-5ab20d024784.png">

Testing with a `restricted_soon` account with no current deadline.

* Update your local server code to return the `account_status` as `restricted_soon` and `current_deadline` as `null` when fetching the basic account info.
* Check the `Payments > Overview` page. You should see no task related to updating account details.

Testing with a `restricted` account when `past_due` is false.

* Update your local server code to return `account_status` as `restricted` and `has_overdue_requirements` as `false`.
* Check the `Payments > Overview` page. You should see no task relating to updating account details.

Testing with a `restricted` account when `past_due` is true.

* Update your local server code to return `account_status` as `restricted` and `has_overdue_requirements` as `true`.
* Check the `Payments > Overview` page. You should see a task saying `Payments and deposits are disabled for this account until missing business information is updated.`

-------------------


**Post merge**
